rules/windows/process_creation/proc_creation_win_csc_susp_parent.yml

```
@@ -18,13 +19,49 @@ logsource:
     category: process_creation
     product: windows
 detection:
-    selection:
-        Image|endswith: '\csc.exe'
+    selection_img:
+        - Image|endswith: '\csc.exe'
+        - OriginalFileName: 'csc.exe'
+    selection_parent_generic:
         ParentImage|endswith:
-            - '\wscript.exe'
             - '\cscript.exe'
             - '\mshta.exe'
-    condition: selection
+            - '\winword.exe'
+            - '\wscript.exe'
+    selection_parent_powershell:
+        ParentImage|endswith:
+            - '\powershell.exe'
+            - '\pwsh.exe'
+        ParentCommandLine|contains:
+            - '-Encoded '
+            - 'FromBase64String'
+    selection_parent_susp_location:
+        - ParentCommandLine|re: '([Pp]rogram[Dd]ata|%[Aa]pp[Dd]ata%|%[Ll]ocal[Aa]pp[Dd]ata%|\[Aa]pp[Dd]ata\([Ll]ocal|[Rr]oaming|[Ll]ocal[Ll]ow)\[^\]{1,26}\.'
```

```suggestion
        - ParentCommandLine|re: '([Pp]rogram[Dd]ata|%([Ll]ocal)?[Aa]pp[Dd]ata%|\\[Aa]pp[Dd]ata\\([Ll]ocal(Ll]ow)?|[Rr]oaming))\\[^\]{1,26}\.'
```
- Maybe also remove the `[^\]{1,26}\.` in the end? It only adds a  literal `.` in the closer proximity. Is it really needed to prevent FPs? (Or what is the intention?)
- Also check my suggestion closely. I made a `\(` to an `\\(` because I think you didn't want to escape the parentheses there. And then I added a matching closing `)` that was missing if my change was what you intended.
- Similar for the escaped `\[` --> `\\[`

---

The idea is to look for files at the first level of those known folders (no sub folder). I think removing the size make sense as its not necessary.